Fix #679, Add Markdown version of OSAL Configuration Guide

ArielSAdamsNASA opened the pull request:
**Describe the contribution**
Pandoc was used to convert the OSAL Configuration Guide to a markdown file. During the conversion, issues were presented such as broken tables. Tables were fixed and block codes were added where appropriate. 

Fix #679

**Additional context**
Please provide feedback on whether the markdown file is acceptable in terms of formatting. 

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

skliper commented:
@astrogeco @ArielSAdamsNASA - what's the status on this?  It would help to move it forward...

skliper commented:
Also delete the other versions... markdown only preferred.

ArielSAdamsNASA commented:
> @astrogeco @ArielSAdamsNASA - what's the status on this? It would help to move it forward...

@skliper Waiting on @acudmore for approval/feedback. I sent him an email this morning.

skliper commented:
Related to #384, the issue to update the contents.  A format check here should be sufficient (and will simplify fixing #384).

astrogeco commented on doc/OSAL-Configuration-Guide-Markdown.md:
```diff
@@ -0,0 +1,798 @@
+[National Aeronautics and Space Administration]{.smallcaps}
```
Remove front matter and have the first section be the Introduction

astrogeco commented on doc/OSAL-Configuration-Guide-Markdown.md:
```diff
@@ -0,0 +1,798 @@
+
+Revision History
+
+| Revision Number |  Release Date | Changes to Prior Revision  | Approval |
```
Maybe move these to an appendix of sorts? I'm not sure we need this when the document is versioned using git

astrogeco commented on doc/OSAL-Configuration-Guide-Markdown.md:
```diff
@@ -0,0 +1,798 @@
+
+[3.1.2 How to Run the OSAL Unit Tests on the PC / Linux Platform 22]
+
+Introduction
```
Use markdown "#" syntax for all headings

astrogeco commented on doc/OSAL-Configuration-Guide-Markdown.md:
```diff
@@ -0,0 +1,798 @@
+
+
+Setting up "classic" build
+--------------------------
```
Use Markdown syntax and get rid of horizontal rule "-----"

astrogeco commented on doc/OSAL-Configuration-Guide-Markdown.md:
```diff
@@ -0,0 +1,798 @@
+
+The RTEMS port of the OSAL has a one to one file system mapping. The OSAL RAM disk will format an RTEMS NVRAM disk with the RFS file system.The file system mappings are defined in the bsp_voltab.c file in the src/bsp/mcf5235-rtems/src directory:
+
+**OSAL File System Mappings**
```
Make this a fourth-level heading "####"

astrogeco commented on doc/OSAL-Configuration-Guide-Markdown.md:
```diff
@@ -0,0 +1,798 @@
+
+To run an example or test, simply do the following:
+
+**\$ sparc-rtems4.10-gdb tasking-example.nxe**
```
Use a code block for this

astrogeco commented on doc/OSAL-Configuration-Guide-Markdown.md:
```diff
@@ -0,0 +1,798 @@
+
+### OSAL Configuration for the PC / Linux Platform
+
+**osal-config.mak Settings**
```
Use Fourth-level headings "####"

astrogeco reviewed (changes requested):
Mainly format cleanups. It might be worthwhile to do a find-and-replace on bold and italic syntax "***" which is sometimes used for subheadings in this document. Similarly horizontal rules "===" are also being used in a subheading fashion. 

It might be worth passing this through a markdown linter like remark or prettier; there are Atom and VSCode plugins for to help with that.

astrogeco reviewed (changes requested):
Remove "markdown" from the file name and use [`git rm` ](https://www.atlassian.com/git/tutorials/undoing-changes/git-rm) to remove the pdf and doc versions of the guide.

ArielSAdamsNASA commented on doc/OSAL-Configuration-Guide.md:
```diff
@@ -0,0 +1,873 @@
+# Introduction
```
Removed front matter so the document begins with the Introduction.

ArielSAdamsNASA commented on doc/OSAL-Configuration-Guide.md:
```diff
@@ -0,0 +1,873 @@
+ | build/unit-tests/osloader-test|Contains the test binary, log, and gcov files for the OSAL loader tests|  
+ |build/unit-tests/ostimer-test|Contains the test binary, log, and gcov files for the OSAL timer tests| 
+
+# Revision History
```
Placed Revision History at the bottom of the document for now.

ArielSAdamsNASA commented:
I removed the pdf and doc versions of the guide and updated the markdown version.

For the markdown version:
- Removed the front matter, so the Introduction is the first section 
- Updated headers to use the markdown syntax # and removed horizontal rule for headers
- Updated table headers to use fourth-level headers 
- Use a code block for the section starting with **\$ sparc-rtems4.10-gdb tasking-example.nxe**
- Moved Revision History to the bottom of the page for now

acudmore commented on doc/OSAL-Configuration-Guide.md:
```diff
@@ -0,0 +1,873 @@
+
+| **Term**    |  **Definition**     |  
+|---|---|
+| Application  (APP)   |  A generic term for a computer program in a
```
This table does not look like it is rendering correctly, at least when viewed on GitHub.
This is what it looks like when I view it.
<img width="556" alt="osal-markdown" src="https://user-images.githubusercontent.com/1873260/102124092-d264f880-3e15-11eb-8d8c-52c34ca2de5e.png">

ArielSAdamsNASA replied:
Updated several tables in terms of formatting. Should be all set to be marked as CCB-READY?

acudmore reviewed (approved):
The conversion is good, there are a couple of table format issues ( in the glossary for example ). But this conversion and removing the old documents is the first step, I think the doc needs to be overhauled in content after this conversion. Much of the technical info is out of date.

astrogeco commented:
Looks good! I would do an interactive rebase and squash the table fix into the original md commit. Also edit the commit message to use the format "Fix #XYZ, COMMIT_DESCRIPTION"

astrogeco commented:
[CCB 2020-12-16](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-December-16-2020) APPROVED

- Separate issue for content
